### DISCUSSION TITLE
using JavaScript to send data to the database and not using the default <form>'s submit

### DISCUSSION THREAD

**Discussion by anjanesh** [alpinejs/alpine#3595] [1. Help] [answered]
I'm using JavaScript to send data to the database and not using the default `<form>`s submit.

```
let g_submitFormData = () =>
{
    const request = new Request(
        BASE_URL + "api/add",
        { headers: {'X-CSRFToken': csrftoken} }
    );

    return fetch(request, {
        method: 'POST',
        mode: 'same-origin',
        body: JSON.stringify(Alpine.store('global_data'))
    })
    .then(response => response.json())
    .then(response =>
    {
        data = response; console.log(data);
        return data;
    });
}
```

But now I am in a fix when I want to try sending in files as Input.

`<input type="file" multiple x-model="$store.global_data.attachments" />`

How do I send in files ?

**Comment by ekwoka** [comment, marked as answer]
idk why you're making a request and then putting that into fetch like that. Fetch already makes it's own request.

But anyway, files have to be added to a `FormData` so you can't stringify it, obviously.

Also you should use async await. :)
